Rules/AvoidPositionalParameters.cs

```diff
@@ -61,9 +62,10 @@ public override IEnumerable<DiagnosticRecord> AnalyzeScript(Ast ast, string file
                 // MSDN: CommandAst.GetCommandName Method
                 if (cmdAst.GetCommandName() == null) continue;
                 
-                if ((Helper.Instance.IsKnownCmdletFunctionOrExternalScript(cmdAst) || declaredFunctionNames.Contains(cmdAst.GetCommandName())) &&
+                if ((Helper.Instance.IsKnownCmdletFunctionOrExternalScript(cmdAst, out CommandInfo commandInfo) || declaredFunctionNames.Contains(cmdAst.GetCommandName())) &&
                     (Helper.Instance.PositionalParameterUsed(cmdAst, true)))
                 {
+                    if (commandInfo?.Parameters.Count == 0) continue;

```

Comment:
should this be done by checking to see if commandInfo is an `ApplicationInfo`? It seems that this is checking for the side effect that an ApplicationInfo won't have parameters

Reply:
Good point
```suggestion
                    if (commandInfo.CommandType == CommandTypes.Application) continue;

```